extensions/wikia/ImageReview/ImageListGetter.class.php

```diff
@@ -0,0 +1,382 @@
+		if ( !empty( $result['error'] ) ) {
+			return $result;
+		}
+		/** @var GlobalFile $imageGlobalFile */
+		$imageGlobalFile = $result['file'];
+		/** @var GlobalTitle $imageGlobalPage */
+		$imageGlobalPage = $result['page'];
+
+		$thumbUrl = $imageGlobalFile->getUrlGenerator()
+			->width( self::IMAGE_REVIEW_THUMBNAIL_SIZE )
+			->height( self::IMAGE_REVIEW_THUMBNAIL_SIZE )
+			->thumbnailDown()
+			->url();
+
+		return [
+			'src' => $thumbUrl,
+			'page' => $imageGlobalPage->getFullURL(),
+			'extension' => $imageGlobalFile->getMimeType(),
+		];
+	}
+
+	/**
+	 * @param $pageId
+	 * @param $wikiId
+	 *
+	 * @return array
+	 *
+	 * @throws Exception
+	 */
+	private function getImageGlobalFile( $wikiId, $pageId ) {
+		try {
+			$imagePage = GlobalTitle::newFromId( $pageId, $wikiId );
+		} catch ( Exception $e ) {
+			return [ 'error' => 'Database for wiki does not exist' ];
+		}
+
+		if ( !$imagePage instanceof GlobalTitle ) {
+			return [ 'error' => 'Page does not exist' ];
+		}
+
+		if ( $imagePage->isRedirect() ) {
+			return [ 'error' => 'Page is a redirect' ];
+		}
+
+		$imageGlobalFile = new GlobalFile( $imagePage );
+		if ( !$imageGlobalFile->exists() ) {
+			return [ 'error' => 'File does not exist' ];
+		}
+
+		return [
+			'file' => $imageGlobalFile,
+		    'page' => $imagePage,
+		];
+	}
+
+	private function moveImagesToIcoStateFromWhere( array $where ) {
+		$count = count( $where );
+		if ( $count <= 0 ) {
+			return;
+		}
+
+		$values = [ 'state' => ImageStates::ICO_IMAGE ];
+		$where = [ implode( 'OR', $where ) ];
+
+		$this->imageStateUpdater->updateBatchImages( $values, $where );
+
+		WikiaLogger::instance()->info( "ImageReviewLog", [
+			'method' => __METHOD__,
+			'message' => "Updated {$count} images (type 'icons')",
+		] );
+	}
+
+	/**
+	 * When assigning images to a user, we move the image from one
+	 * state to another to indicate it's been assigned. If we're assigning
+	 * an "unreviewed" image it moves into the "reviewed" state. If we're
+	 * assigning a "rejected" image it moves into the "rejected_in_review"
+	 * state. And if we're assigning a "questionable" image it moves into
+	 * the "questionable_in_review" state.
+	 * @return int
+	 */
+	private function getNewState() : int {
+		if ( $this->state == ImageStates::REJECTED ) {
+			return ImageStates::REJECTED_IN_REVIEW;
+		}
+
+		if ( $this->state == ImageStates::QUESTIONABLE ) {
+			return ImageStates::QUESTIONABLE_IN_REVIEW;
+		}
+
+		return ImageStates::IN_REVIEW;
+	}
+
+	private function fetchedEnoughImages() : bool {
+		return count( $this->imageList ) == self::LIMIT_IMAGES;
+	}
+
+	/**
+	 * Setup the labels used to indicated different states for each image
+	 * @return array
+	 */
+	private function getLabelValues() {
+		$labels = [];
+
+		$labels['labelOk'] = [
+			'value' => ImageStates::APPROVED,
+			'checked' => in_array( $this->state, [ImageStates::APPROVED, ImageStates::IN_REVIEW, ImageStates::UNREVIEWED] ) ? 'checked' : ''
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "checked" 6 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)